dynamo/src/main/java/no/bibsys/db/TableDriver.java

```diff
@@ -78,80 +73,77 @@ public boolean tableExists(final String tableName) {
         return exists;
     }
 
-    
+
     /**
      * Return number of items in table
      * 
      * @param tableName
      * @return number of items
      */
-    
+
     public long tableSize(final String tableName) {
-    
-        return getTable(tableName).describe().getItemCount(); // Only updated every 6 hours!
+
+        ScanRequest scanRequest = new ScanRequest(tableName).withSelect(Select.COUNT);
+        ScanResult result = client.scan(scanRequest);
+        Integer itemCount = result.getScannedCount();
+        return itemCount;
     }
 
-    /**
-     * Deletes a table and the validation schema associated with this table
-     */
-    public void deleteTable(final String tableName) throws InterruptedException {
-        long itemCount = dynamoDb.getTable(tableName).describe().getItemCount();
+    public boolean deleteTable(final String tableName) {
+        
+        if(!tableExists(tableName)) {
+            return false;
+        }
+        ScanRequest scanRequest = new ScanRequest(tableName).withSelect(Select.COUNT);
```

Comment:
pack the checking if the table is empty in a function that is called isEmpty(), so that it is clear what we check